src/main/java/io/lighty/yang/validator/formats/Analyzer.java

```diff
@@ -28,13 +31,30 @@
 
     @Override
     void emitFormat() {
-        for (Object subStatement : ((EffectiveSchemaContext) this.schemaContext).getRootDeclaredStatements()) {
-            analyzeSubstatement((DeclaredStatement) subStatement);
+        Set<DeclaredStatement<?>> declaredStatements = new HashSet<>();
+        fillDeclaredStatements(declaredStatements, this.schemaContext.getModules());
+        for (DeclaredStatement<?> declaredStatement : declaredStatements) {
+            analyzeSubstatement(declaredStatement);
         }
-
         printOut();
     }
 
+    private void fillDeclaredStatements(final Set<DeclaredStatement<?>> resultStatements,
+            final Collection<? extends Module> modules) {
+        for (Module module : modules) {
+            resultStatements.add(((EffectiveStatement<?, ?>) module).getDeclared());
+
+            Collection<? extends Module> submodules = module.getSubmodules();
+            if (submodulesAreNotEmpty(submodules)) {
+                fillDeclaredStatements(resultStatements, submodules);
```

Comment:
after output param will be removed, you can recursively addAll(fillDeclaredStatements())